include/pybind11/pybind11.h

```diff
@@ -1377,6 +1377,14 @@ template <typename Type> class enum_ : public class_<Type> {
             return m;
         }, return_value_policy::copy);
         def(init([](Scalar i) { return static_cast<Type>(i); }));
+        def(init([name, m_entries_ptr](std::string value) -> Type {
+            pybind11::dict values = reinterpret_borrow<pybind11::dict>(m_entries_ptr);
```

Comment:
This isn't your code, but let's clean it up anyway: instead of `auto m_entries_ptr = m_entries.inc_ref().ptr();` being a raw, incremented pointer, we can change it to: `auto entries = m_entries;`.  Then all the lambdas (this, plus the others that do a `reinterpret_borrow`) can just capture `entries` and can simplify their `reinterpret_borrow<dict>(m_entries_ptr)` to just `entries`.

Reply:
Fixed, much nicer the new way. Thanks!